code/modules/projectiles/special.dm

```diff
@@ -963,3 +963,118 @@ ABSTRACT_TYPE(/datum/projectile/special)
 	on_hit(atom/hit, direction, projectile)
 		explosion_new(projectile, get_turf(hit), explosion_power, 1)
 		..()
+
+/datum/projectile/special/shotchem // how do i shot chem
+	name = "chemical bolt"
+	sname = "chembolt"
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "extinguish"
+	shot_sound = "sound/effects/spray2.ogg"
+	power = 0
+	cost = 1
+	damage_type = D_SPECIAL
+	shot_delay = 1 DECI SECOND
+	dissipation_rate = 0
+	dissipation_delay = 0
+	ks_ratio = 0
+	hit_ground_chance = 0 // burn right over em
+	max_range = 10
+	silentshot = 1 // Mr. Muggles is hit by the chemical bolt x99999
+	fullauto_valid = 0
+
+
+	/// Releases some of the projectile's gas into the turf
+	proc/emit_gas(turf/T, all_of_it = 0)
+		if(!src.master || !src.master.special_data)
+			return
+		var/datum/gas_mixture/airgas = src.master.special_data["airgas"]
+		T?.assume_air(airgas.remove_ratio(all_of_it ? 1 : src.master.special_data["chem_pct_app_tile"]))
+
+	/// Sprays some of the chems in the projectile onto everything on hit's turf
+	/// Try not to pass it something with an organholder, acid-throwers will disintigrate all their organs
+	proc/emit_chems(atom/hit, obj/projectile/O, angle)
+		if(!O.special_data || !length(O.special_data) || !istype(hit) || !O.reagents)
+			return
+
+		var/turf/T = get_turf(hit)
+
+		var/list/c_turfH = O.special_data["crossed_turfs"]
+		if (c_turfH[src.turf2coordinates(T)] == 1)
+			return // one spray per turf, please! No matter how many times this gets called!
+		c_turfH[src.turf2coordinates(T)] = 1
+
+		var/datum/reagents/chemR = O.reagents
+		var/chem_amt = chemR.total_volume
+		/// If there's just a little bit left, use the rest of it
+		var/amt_to_emit = (chem_amt <= 0.1) ? chem_amt : (chemR.maximum_volume * O.special_data["chem_pct_app_tile"])
+
+		var/datum/reagents/copied = new/datum/reagents(amt_to_emit)
+		copied = chemR.copy_to(copied, 1, copy_temperature = 1)
+
+		if(!T.reagents) // first get the turf
+			T.create_reagents(100)
+		copied.copy_to(T.reagents, 1, copy_temperature = 1)
+		copied.reaction(T, TOUCH, 0, 0)
+		if(O.special_data["IS_LIT"]) // Heat if needed
+			T.reagents?.set_reagent_temp(O.special_data["burn_temp"], TRUE)
+		for(var/atom/A in T.contents) // then all the stuff in the turf
+			if(istype(A, /obj/overlay) || istype(A, /obj/projectile))
+				continue
+			copied.reaction(A, TOUCH, 0, 0)
+			if(!A.reagents)
+				A.create_reagents(100)
+			copied.copy_to(A.reagents, 1, copy_temperature = 1)
+			if(O.special_data["IS_LIT"]) // Heat if needed
+				A.reagents.set_reagent_temp(O.special_data["burn_temp"], TRUE)
```

Comment:
I feel like a sweeping rework like this is perhaps a good time to get rid of flamers coping into mobs/objects, and only apply the touch reactions (including possible skin penetration) to those

Reply:
I dunno about that. Flamers have always done that, and in a way became one of its main features. Though, yeah, it could be used to dupe chems, many chemnerds have come to expect it to do... whatever it is chemnerds do with it.

Reply:
fuk da nerds

Reply:
indeed